### PR TITLE
Enchancement - Add support for ioredis Cluster object 2nd parameter

### PR DESCRIPTION
**Pull request by scvgoe**
It comes from #3538

### CODE REVIEW

File: src/cache/RedisQueryResultCache.ts [hunk 1]
```diff
@@ -60,14 +60,12 @@ export class RedisQueryResultCache implements QueryResultCache {
             } else {
                 this.client = new this.redis();
             }
-
         } else if (this.clientType === "ioredis/cluster") {
-            if (cacheOptions && cacheOptions.options) {
-                this.client = new this.redis.Cluster(cacheOptions.options);
+            if (cacheOptions && cacheOptions.options && cacheOptions.options.startupNodes) {
```
Comment (vlapo) [inline src/cache/RedisQueryResultCache.ts]: `options` argument is not required according to source code from `ioredis` so do not have to check `cacheOptions.options` existence.
https://github.com/luin/ioredis/blob/fbcf692f5de8c653139e4b053c6356ab6d67e463/lib/cluster/index.ts#L65

**Comment by vlapo** [comment]
First at all thank you for contributing. We are appreciate it. I have one inline comment and please also add record to `CHANGELOG.md`.

@HussainAliAkbar, @ignacioblit if you have any comments.

File: src/cache/RedisQueryResultCache.ts (hunk 1 above)
Comment (ignacioblit) [reply src/cache/RedisQueryResultCache.ts]: Nice enhancement @scvgoe!

This looks good to me. The optional value is actually cacheOptions.options.options.  

cacheOptions.options is the typeORM options object with the general Cluster Ioredis properties including the mandatory one: startupNodes. So in the case of Cluster Redis its always required. 

I have to say that: cacheOptions.options.options looks pretty weird and maybe confusing, as it has plenty of options. Though I have no better suggestion than changing the name of the property to clusterOptions. Yet this is none important, so ignore this if you want :)

Comment (vlapo) [reply src/cache/RedisQueryResultCache.ts]: Sry man :-) did not realize it because of so much "options".

Comment (ignacioblit) [reply src/cache/RedisQueryResultCache.ts]: Hahaha, exactly!

Comment (vlapo) [reply src/cache/RedisQueryResultCache.ts]: So just add recordt to `CHANGELOG.md` and it is ready to merge.

**Comment by HussainAliAkbar** [comment]
This looks good to me :+1: Nice enhancement indeed!

File: src/cache/RedisQueryResultCache.ts (hunk 1 above)
Comment (scvgoe) [reply src/cache/RedisQueryResultCache.ts]: Thank you for your reviewing!!

**Comment by scvgoe** [comment]
When will be released this pr?

**Comment by scvgoe** [comment]
Graet idea!

File: docs/caching.md [hunk 2]
```diff
@@ -158,10 +158,46 @@ In case you want to connect to a redis-cluster using IORedis's cluster functiona
...
+                clusterRetryStrategy: function (times) { return null },
+                redisOptions: {
+                    maxRetriesPerRequest: 1
+                }
+            }
+        }
+    }
+}
+```
+Just specify all the nodes in the cluster inside an array with their hosts and ports.
```
Comment (vlapo) [inline docs/caching.md]: This sentence after object config is really confusing. Also object based configuration should be first (as it provide more options) and array based config should be only as simple note with description.

**Comment by pleerock** [comment]
Thank you!

**Pull request merged**